server/models.py

```diff
@@ -1643,3 +1644,93 @@ class CanvasAssignment(Model):
     @property
     def url(self):
         return '{}/assignments/{}'.format(self.canvas_course.url, self.external_id)
+
+#########
+# Files #
+#########
+
+class ExternalFile(Model):
+
+    id = db.Column(db.Integer, primary_key=True)
+    # Bucket/Folder that the file is stored in
+    container = db.Column(db.String(255), nullable=False)
+    filename = db.Column(db.String(1024), nullable=False)
+    object_name = db.Column(db.String(1024), nullable=False)
+
+    is_staff = db.Column(db.Boolean, nullable=False)
```

Comment:
Even in the case where a student gets a staff member to be in their group I think we should still offer some level of protection. (or we should disallow staff members from joining a group with students) 

The other reason to have it is a slightly easier query for all student generated files for a particular assignment. Without it, you could just instead do a query where user.id in student_ids. 

Renaming sounds fine (`staff_file`? `for_staff`?) . Let me know if you want to just remove it now or remove it after we disallow staff members from joining student groups.